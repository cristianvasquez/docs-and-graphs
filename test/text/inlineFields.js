import { expect } from 'expect'

import toMatchSnapshot from 'expect-mocha-snapshot'
import {
  extractInlineFields, removeInlineFields,
} from '../../src/text/inlineFields.js'

const markdown = [
  'inline :: field',
  'nested ( inline :: field )',
  '**Thoughts**:: It was decent.',
  '**Rating**:: 6',
  '[mood:: okay] | [length:: 2 hours]',
  'No inline fields',
]

expect.extend({ toMatchSnapshot })

describe('extractInlineFields', async function () {
  for (const current of markdown) {
    it(current, async function () {

      const fields = extractInlineFields(current)

      expect(fields).toMatchSnapshot(this)
    })
  }
})

describe('removeInlineFields', async function () {
  for (const current of markdown) {
    it(current, async function () {

      const text = removeInlineFields(current)

      expect(text).toMatchSnapshot(this)
    })
  }
})
