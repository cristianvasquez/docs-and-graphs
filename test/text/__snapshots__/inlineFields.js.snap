// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractInlineFields **Rating**:: 6 1`] = `
Array [
  Object {
    "property": "**Rating**",
    "raw": "**Rating**:: 6",
    "value": " 6",
  },
]
`;

exports[`extractInlineFields **Thoughts**:: It was decent. 1`] = `
Array [
  Object {
    "property": "**Thoughts**",
    "raw": "**Thoughts**:: It was decent.",
    "value": " It was decent.",
  },
]
`;

exports[`extractInlineFields [mood:: okay] | [length:: 2 hours] 1`] = `
Array [
  Object {
    "property": "mood",
    "raw": "[mood:: okay]",
    "value": " okay",
  },
  Object {
    "property": "length",
    "raw": "[length:: 2 hours]",
    "value": " 2 hours",
  },
]
`;

exports[`extractInlineFields No inline fields 1`] = `Array []`;

exports[`extractInlineFields inline :: field 1`] = `
Array [
  Object {
    "property": "inline ",
    "raw": "inline :: field",
    "value": " field",
  },
]
`;

exports[`extractInlineFields nested ( inline :: field ) 1`] = `
Array [
  Object {
    "property": " inline ",
    "raw": "( inline :: field )",
    "value": " field ",
  },
]
`;

exports[`removeInlineFields **Rating**:: 6 1`] = `""`;

exports[`removeInlineFields **Thoughts**:: It was decent. 1`] = `""`;

exports[`removeInlineFields [mood:: okay] | [length:: 2 hours] 1`] = `" | "`;

exports[`removeInlineFields No inline fields 1`] = `"No inline fields"`;

exports[`removeInlineFields inline :: field 1`] = `""`;

exports[`removeInlineFields nested ( inline :: field ) 1`] = `"nested "`;
